services/vpc_manager/src/main/java/com/futurewei/alcor/vpcmanager/service/Impl/VpcServiceImpl.java

```diff
@@ -102,4 +108,19 @@ public boolean checkSubnetsAreEmpty(VpcEntity vpcEntity) throws SubnetsNotEmptyE
         }
         return true;
     }
+
+    @Override
+    public ResponseId registerVpc(VpcEntity vpcEntity) {
+        String url = gatewayUrl + "project/" + vpcEntity.getProjectId() + "/gatewayinfo";
+        VpcInfo vpcInfo = new VpcInfo(vpcEntity.getId(), vpcEntity.getSegmentationId(), vpcEntity.getProjectId());
+        HttpEntity<VpcInfoJson> vpcHttpEntity = new HttpEntity<>(new VpcInfoJson(vpcInfo));
+        return restTemplate.postForObject(url, vpcHttpEntity, ResponseId.class);
+    }
+
+    @Override
+    public ResponseId unRegisterVpc(VpcEntity vpcEntity) {
+        String url = gatewayUrl + "project/" + vpcEntity.getProjectId() + "/gatewayinfo/" + vpcEntity.getId();
+        restTemplate.delete(url);
+        return new ResponseId(vpcEntity.getId());
```

Comment:
@songxiaoyan did you handle workflow for unregister vpc in gateway manager which is sending notification to zeta and dpm?

Reply:
and remove attachment and gateway entry in gateway manager.

Reply:
I already did it in the GatewayManager, you can review code in this PR.

Reply:
I already did it in the GatewayManager, you can review code in this PR.